internal/auth/claims_test.go

```diff
@@ -0,0 +1,103 @@
+package auth_test
+
+import (
+	"crypto/rsa"
+	"fmt"
+	"testing"
+
+	"github.com/ardanlabs/service/internal/auth"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+func TestParseClaims(t *testing.T) {
+	// Parse the private key used to generate the token.
+	prvKey, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(privateRSAKey))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Generate the token.
+	signedClaims := auth.Claims{
+		Roles: []string{auth.RoleAdmin},
+	}
+	tknStr, err := auth.GenerateToken(prvKey, privateRSAKeyID, signedClaims)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Parse the public key used to validate the token.
+	pubKey, err := jwt.ParseRSAPublicKeyFromPEM([]byte(publicRSAKey))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	parser := auth.NewParser(singleKeyFunc(privateRSAKeyID, pubKey))
+
+	parsedClaims, err := parser.ParseClaims(tknStr)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Assert expected claims.
+	if exp, got := len(signedClaims.Roles), len(parsedClaims.Roles); exp != got {
+		t.Fatalf("expected %v roles, got %v", exp, got)
+	}
+	if exp, got := signedClaims.Roles[0], parsedClaims.Roles[0]; exp != got {
+		t.Fatalf("expected roles[0] == %v, got %v", exp, got)
+	}
+}
+
+func singleKeyFunc(id string, key *rsa.PublicKey) auth.KeyFunc {
```

Comment:
Unless it is used more than once, I could be missing that, why not just assign this to a variable in the test and use it.